os/compression/compress_read.c

```diff
@@ -404,9 +401,18 @@ void compress_uninit(void)
 {
 #if CONFIG_COMPRESSION_TYPE == 1
 	/* Freeing memory allocated to read_buffer and out_buffer for file decompression */
-	if (compression_header.compression_format == COMPRESSION_TYPE_LZMA) {
-		kmm_free(buffers.read_buffer);
-		kmm_free(buffers.out_buffer);
+	if (compression_header->compression_format == COMPRESSION_TYPE_LZMA) {
```

Comment:
Is it okay not to check that compression_header is NULL or not?

Reply:
1. compress_uninit is called only if compress_init is successful (i.e. kmm_malloc for 
    compression_header is success).
    So no need for NULL check here.
2. `#if CONFIG_COMPRESSION_TYPE == 1` is present in both compress_init/compress_uninit 
    for  buffers. This ensures this specific memory allocation/deallocation logic is compiled only 
    for `LZMA algorithm`.
    For some other decompression algorithm, some other logic can be added under
    `#elif CONFIG_COMPRESSION_TYPE == 2`.